hopsworks-api/src/main/java/io/hops/hopsworks/api/jupyter/JupyterService.java

```diff
@@ -332,12 +332,18 @@ public Response startNotebookServer(JupyterSettings jupyterSettings, @Context Ht
 
       String externalIp = Ip.getHost(req.getRequestURL().toString());
 
-      jp = jupyterFacade.saveServer(externalIp, project, configSecret,
+      try {
+        jp = jupyterFacade.saveServer(externalIp, project, configSecret,
           dto.getPort(), user.getId(), dto.getToken(), dto.getPid());
+      } catch(Exception e) {
+        stopAllSessions(hdfsUser, hopsworksUser, configSecret, dto.getPid(), dto.getPort());
```

Comment:
@robzor92 We should log the exception here.

Reply:
Fixed